aws/resource_aws_dynamodb_table.go

```diff
@@ -250,6 +265,16 @@ func resourceAwsDynamoDbTableCreate(d *schema.ResourceData, meta interface{}) er
 		}
 	}
 
+	if v, ok := d.GetOk("encrypt_at_rest"); ok {
+		options := v.([]interface{})
+		if options[0] == nil {
```

Comment:
I think this needs to check `len(options) == 0`

```go
s := make([]interface{}, 0)
if s[0] == nil {
	log.Println("All is well")
}

# panic: runtime error: index out of range

s := make([]interface{}, 0)
if len(s) == 0 {
	log.Println("All is well")
}

# 2018/03/01 15:19:00 All is well
```

Reply:
Generally speaking though, we forego error checking against our own schema.

Reply:
Doesn't `GetOk()` guard against `len(options) == 0`?